app/Domain/Coupons/Coupon.php

```diff
@@ -10,8 +10,10 @@ final class Coupon
 {
     public function __construct(
         public readonly UuidInterface $id,
+        public readonly bool $isDistributed,
         public readonly ?UuidInterface $integrationId,
-        public readonly string $code
+        public readonly string $code,
+        public readonly CouponStatus $status,
```

Comment:
This field can be derived from `isDistributed` and `integrationId` so I would not make it a constructor argument. If you make it an argument this could lead to wrong combinations.